boards/arm/v2m_beetle/CMakeLists.txt

```diff
@@ -1,7 +1,5 @@
 # SPDX-License-Identifier: Apache-2.0
 
-if(CONFIG_PINMUX_BEETLE)
-  zephyr_library()
-  zephyr_library_include_directories(${ZEPHYR_BASE}/drivers)
-  zephyr_library_sources(pinmux.c)
-endif()
+zephyr_library()
+zephyr_library_include_directories(${ZEPHYR_BASE}/drivers)
```

Comment:
this looks suspicious, can likely be removed

Reply:
yeah I tried, fails the build on `#include <gpio/gpio_cmsdk_ahb.h>`, somehow it's including the internal header, should be split out in `include/drivers`, won't bother for this PR :-)

Reply:
Well I added a patch for that.